torchmetrics/audio/pesq.py

```diff
@@ -0,0 +1,132 @@
+# Copyright The PyTorch Lightning team.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from typing import Any, Callable, Optional
+
+from torch import Tensor, tensor
+
+from torchmetrics.functional.audio.pesq import pesq
+from torchmetrics.metric import Metric
+from torchmetrics.utilities.imports import _PESQ_AVAILABLE
+
+
+class PESQ(Metric):
+    """PESQ (Perceptual Evaluation of Speech Quality)
+
+    This is a wrapper for the pesq package [1]. . Note that input will be moved to `cpu`
+    to perform the metric calculation.
+
+    .. note:: using this metrics requires you to have ``pesq`` install. Either install as ``pip install
+        torchmetrics[audio]`` or ``pip install pesq``
+
+    Forward accepts
+
+    - ``preds``: ``shape [...,time]``
+    - ``target``: ``shape [...,time]``
+
+    Args:
+        fs:
+            sampling frequency, should be 16000 or 8000 (Hz)
+        mode:
+            'wb' (wide-band) or 'nb' (narrow-band)
+        keep_same_device:
+            whether to move the pesq value to the device of preds
+        compute_on_step:
+            Forward only calls ``update()`` and return ``None`` if this is set to ``False``.
+        dist_sync_on_step:
+            Synchronize metric state across processes at each ``forward()``
+            before returning the value at the step
+        process_group:
+            Specify the process group on which synchronization is called.
+            default: ``None`` (which selects the entire world)
+        dist_sync_fn:
+            Callback that performs the allgather operation on the metric state. When ``None``, DDP
+            will be used to perform the allgather
+
+    Raises:
+        ValueError:
+            If ``peqs`` package is not installed
+        ValueError:
+            If ``fs`` is not either  ``8000`` or ``16000``
+        ValueError:
+            If ``mode`` is not either ``"wb"`` or ``"nb"``
+
+    Example:
+        >>> from torchmetrics.audio import PESQ
+        >>> import torch
+        >>> g = torch.manual_seed(1)
+        >>> preds = torch.randn(8000)
+        >>> target = torch.randn(8000)
+        >>> nb_pesq = PESQ(8000, 'nb')
+        >>> nb_pesq(preds, target)
+        tensor(2.2076)
+        >>> wb_pesq = PESQ(16000, 'wb')
+        >>> wb_pesq(preds, target)
+        tensor(1.7359)
+
+    References:
+        [1] https://github.com/ludlows/python-pesq
+    """
+
+    sum_pesq: Tensor
+    total: Tensor
+
```

Comment:
```suggestion
    total: Tensor
    is_differentiable = False
    higher_is_better = True

```

Reply:
we did some updates recently were `is_differentiable` is moved to a class attribute and `higher_is_better` is introduced to indicate if the metric is best when maximized/minimized

Reply:
Yes, higher is better